### DISCUSSION TITLE
Trigger idea

### DISCUSSION THREAD

**Discussion by gzqn** [Budibase/budibase#4384] [Ideas]
From the docs I believe it is not possible to click on a field and initiate an action. My use case would be to display a link in a field which when the field is clicked would then do things like: 

-  open a folder (ie in windows file explorer)  ie providing direct access to multiple changing resources or 
- file (ie open an application and display a specific file) or 
- open an external file and display in bb app (I presume this is how attachments display though I haven't got far enough to test this :(,  

being able to open an external file makes updating what would otherwise be attachments more flexible and not subject to IT support for people who otherwise shouldn't have access to the backend.  Being taken directly to a location means the user can directly access information that is not usefully stored in bb. This would take user workflow out of bb and extend its useability.

I presume this would require a new column type. There would be security implications, so file types would need limiting and scripts prevented from running. 

There are use cases for running scripts that are external to bb, but this would be best done from an additional column type which at the least provides the user with a warning, a description and check of where the path originates (ie is it an authorised network location). For an app which has a user  base external to the bb authoring organisation, this should be restricted to scripts contained within the BB application, and an option to run the script through a virus checker. There are probably other security issues that i do not understand.

**Comment by mjashanks** [comment]
I'm pretty sure this will never be possible. As far as I know browsing the local directory is not allowed via the web browser. You can link to local files like `<a href="file:///......">` but the browser will only open certain types of file - i.e. HTML, JS etc.
